src/core/lib/channel/promise_based_filter.h

```diff
@@ -1903,15 +1909,16 @@ struct ChannelFilterWithFlagsMethods {
     auto status = F::Create(args->channel_args,
                             ChannelFilter::Args(args->channel_stack, elem));
     if (!status.ok()) {
-      static_assert(
-          sizeof(promise_filter_detail::InvalidChannelFilter) <= sizeof(F),
-          "InvalidChannelFilter must fit in F");
-      new (elem->channel_data) promise_filter_detail::InvalidChannelFilter();
+      new (elem->channel_data) F*(nullptr);
       return absl_status_to_grpc_error(status.status());
     }
-    new (elem->channel_data) F(std::move(*status));
+    new (elem->channel_data) F*(status->release());
```

Comment:
Just calling out, not really a problem: This means `shared_ptr<>` wouldn't work.

Reply:
Yup. I think that's ok.
(if we need it we can bring in a new specialization, but hopefully this code is deleted by then)